go/tendermint/tendermint.go

```diff
@@ -357,7 +357,11 @@ func (t *tendermintService) Subscribe(subscriber string, query tmpubsub.Query) (
 	// force-unsubscribe the channel if processing takes too long.
 
 	subFn := func() (tmtypes.Subscription, error) {
-		return t.node.EventBus().SubscribeUnbuffered(t.ctx, subscriber, query)
+		sub, err := t.node.EventBus().SubscribeUnbuffered(t.ctx, subscriber, query)
+		if err != nil {
+			return nil, err
+		}
+		return newTendermintPubsubBuffer(sub), nil
```

Comment:
Don't forget to update `broadcastTx` to remove the "drain hack" that shouldn't be needed anymore.